File: Suhyun/Level1/1월 2주차/자연수 뒤집어 배열로 만들기/Reverse_number.java
import java.util.Arrays;

public class Reverse_number {
    public static void main(String[] args) {
        long n=123123123;
        int num=(int)(Math.log10(n)+1);
        int[] answer=new int[num];
        
        for(int i=0; i<num; i++){
            answer[i]=(int)(n%10);
            n=n/10;
        }

        System.out.println(Arrays.toString(answer));
        //return asnwer;
    }
}

File: Suhyun/Level1/1월 2주차/자연수 뒤집어 배열로 만들기/Reverse_number.js
let n=123456789;
let str=n.toString();
let ch=str.split('');

let answer=[];

for(let i=str.length; i>0 ; i--){
    let num=parseInt(ch[i-1]);
    answer.push(num);

    //answer.push(parseInt(ch[i-1]));


}
console.log(answer);

//프로그래머스용 
function solution(n) {
    let str=n.toString();
    let ch=str.split('');
    let answer=[];
        
    for(let i=str.length; i>0 ; i--){
        let num=parseInt(ch[i-1]);
        answer.push(num);
    
    }
        return answer;
    }

// 속도는 비슷하나 깔끔
return n.toString().split('').reverse().map(o => o = parseInt(o));